microsoft/testsuites/vm_extensions/linux_patch_extension.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+from assertpy.assertpy import assert_that
+from lisa.base_tools.service import Service
+from lisa.sut_orchestrator.azure.common import (
+    get_compute_client,
+    get_node_context,
+    wait_operation,
+)
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.util import LisaException
+
+def _verify_vm_agent_running(node: Node, log: Logger) -> None:
+    service = node.tools[Service]
+    is_vm_agent_running = service.is_service_running("walinuxagent.service") or service.is_service_running("waagent.service")
+    
+    if is_vm_agent_running:
+        log.debug("verify walinuxagent or waagent is running")
+        assert_that(is_vm_agent_running).described_as(
+            "Expected walinuxagent or waagent is running"
+        ).is_true()
+    else:
+        log.debug("verify walinuxagent or waagent is not running")
+        raise LisaException("walinuxagent or waagent is not running. Test case failed.")
```

Comment:
You don't need to add "test case failed". If the exception is raised, it means the test case failed.